library/src/main/java/com/chuckerteam/chucker/internal/support/TeeSource.kt

```diff
@@ -83,22 +83,23 @@ internal class TeeSource(
         if (!isFailure) {
             isFailure = true
             sideStream.close()
-            callback.onFailure(exception, sideChannel)
+            callback.onFailure(sideChannel, exception)
         }
     }
 
     interface Callback {
         /**
          * Called when the upstream was closed. All read bytes are copied to the [file].
-         * This does not mean that the content of a [file] is valid. Only that the user
-         * is done with the reading process.
+         * This does not mean that the content of the [file] is valid. Only that the user
```

Comment:
Minor observation - I guess `user` on this line and `client` on 95 mean the same thing. Suggest to pick one word for naming. `client` seems more suitable for this case.